groups/ntc/ntcs/ntcs_dispatch.cpp

```diff
@@ -1884,7 +1884,7 @@ void Dispatch::announceDetached(
     const bsl::shared_ptr<ntci::ProactorSocket>& socket,
     const bsl::shared_ptr<ntci::Strand>&         destination)
 {
-    if (!destination) {
+    if (NTCCFG_LIKELY(!destination)) {
```

Comment:
One of the practices which seems to be overuse in this library is usage of compiler prediction hints everywhere. And I think that this is one example of that. 
According to my understanding we should keep thinking about dynamic load balancing as well, and in case of dynamic load balancing usage (as it is implemented right now) this hint wil be totally wrong.

Reply:
I will remove the prediction hints here.